groovy-language-server/src/test/java/com/palantir/groovylanguageserver/GroovycWrapperTest.java

```diff
@@ -379,18 +383,589 @@ public void testCompile_error() throws InterruptedException, ExecutionException,
+        assertEquals(0, wrapper.findReferences(createReferenceParams(file.getAbsolutePath(), 8, 1, false)).size());
+        // Right after "Cat", therefore should not find any symbol
+        assertEquals(0, wrapper.findReferences(createReferenceParams(file.getAbsolutePath(), 11, 3, false)).size());
+
+        // InnerCat2 references - testing finding more specific symbols that are contained inside another symbol's
+        // range.
+        assertEquals(Sets.newHashSet(
+                createSymbolInformation("myFriend", file.getAbsolutePath(),
+                        SymbolKind.Field, 13, 4, 13, 22, Optional.of("Cat2")),
+                createSymbolInformation("getMyFriend", file.getAbsolutePath(),
+                        SymbolKind.Method, -1, -1, -1, -1, Optional.of("Cat2")),
+                createSymbolInformation("value", file.getAbsolutePath(),
+                        SymbolKind.Variable, -1, -1, -1, -1, Optional.of("setMyFriend"))),
+                wrapper.findReferences(createReferenceParams(file.getAbsolutePath(), 14, 10, false)));
+        // Find one line enum correctly
+        assertEquals(Sets.newHashSet(
+                createSymbolInformation("ONE", enumFile.getAbsolutePath(),
+                        SymbolKind.Field, 1, 14, 1, 17, Optional.of("MyEnum")),
+                createSymbolInformation("TWO", enumFile.getAbsolutePath(),
+                        SymbolKind.Field, 1, 18, 1, 21, Optional.of("MyEnum"))),
+                wrapper.findReferences(createReferenceParams(enumFile.getAbsolutePath(), 1, 7, false)).stream()
+                        .filter(symbol -> Ranges.isValid(symbol.getLocation().getRange())).collect(Collectors.toSet()));
+        // Identify type A correctly
+        assertEquals(Sets.newHashSet(
+                createSymbolInformation("a", innerClass.getAbsolutePath(),
+                        SymbolKind.Field, 2, 1, 2, 4, Optional.of("A"))),
+                wrapper.findReferences(createReferenceParams(innerClass.getAbsolutePath(), 1, 7, false)).stream()
+                .filter(symbol -> Ranges.isValid(symbol.getLocation().getRange())).collect(Collectors.toSet()));
+        // Identify type B correctly
+        assertEquals(Sets.newHashSet(
+                createSymbolInformation("b", innerClass.getAbsolutePath(),
+                        SymbolKind.Field, 3, 1, 3, 4, Optional.of("A"))),
+                wrapper.findReferences(createReferenceParams(innerClass.getAbsolutePath(), 1, 18, false)).stream()
+                .filter(symbol -> Ranges.isValid(symbol.getLocation().getRange())).collect(Collectors.toSet()));
+    }
+
+    @Test
+    public void testReferences_classesAndInterfaces() throws InterruptedException, ExecutionException, IOException {
+        File newFolder1 = root.newFolder();
+        File extendedCoordinatesFile = addFileToFolder(newFolder1, "ExtendedCoordinates.groovy",
+                        "class ExtendedCoordinates extends Coordinates{\n"
```

Comment:
The problem is a lot of them are not exactly the same, which would lead to a whole bunch of files like `Coordinates_testReferences_classesAndInterfaces.groovy`.